lockfile/src/javascript.rs

```diff
@@ -356,16 +356,23 @@ impl PnpmLock {
         let name = key.split_once('(').map(|(name, _)| name).unwrap_or(key);
 
         // Get version separator based on PNPM version.
-        let version_separator = match version {
+        let version_separator = match pnpm_version {
```

Comment:
Not exactly related to this PR, but should the supported PNPM lockfile formats be listed in `docs/supported_lockfiles.md`? We do that for a few other lockfiles that have multiple recognized formats.

Reply:
Probably? Though we do support a wide enough version range to where I've yet to find a lockfile that we cannot parse (other than bugs like this).